src/librustc_infer/infer/higher_ranked/mod.rs

```diff
@@ -65,9 +65,8 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     ///
     /// **Important:** you must call this function from within a snapshot.
     /// Moreover, before committing the snapshot, you must eventually call
-    /// either `plug_leaks` or `pop_placeholders` to remove the placeholder
-    /// regions. If you rollback the snapshot (or are using a probe), then
-    /// the pop occurs as part of the rollback, so an explicit call is not
+    /// do stuff to prevent problems (TODO). If you rollback the snapshot (or are using a probe),
+    /// then the pop occurs as part of the rollback, so an explicit call is not
```

Comment:
This still has to be fixed, but I don't know what exactly has to be done here. Started a discussion on zulip: https://rust-lang.zulipchat.com/#narrow/stream/144729-wg-traits/topic/Placeholder.20invariants